### ISSUE TITLE
Parse LLVM failed

### ISSUE THREAD

**Issue by csordasmarton** [Ericsson/CodeCompass#117]
Parsing LLVM with CodeCompass has been failed with the following message:
```
[INFO] (1254/2425) Parsing /ssd/<username>/projects/llvm-3.8.0.src/lib/ProfileData/CoverageMappingWriter.cpp
warning: unknown warning option '-Wno-maybe-uninitialized'; did you mean '-Wno-uninitialized'? [-Wunknown-warning-option]
CodeCompass_parser: /ssd/<username>/flash-deps/deps/llvm-3.8.0.src/tools/clang/include/clang/Lex/Token.h:128: unsigned int clang::Token::getLength() const: Assertion `!isAnnotation() && "Annotation tokens have no length field"' failed.
```

**Comment by whisperity** [comment]
**Issue appears to be related to the macro expansion.**

This issue existed when we were using LLVM 3.6 as CodeCompass' backend. The project I parsed is the "latest" master LLVM somewhere **as of Friday noon**, but it looks like this issue has been around for quite a while.

```
#0  0x00007ffff5422428 in __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:54
#1  0x00007ffff542402a in __GI_abort () at abort.c:89
#2  0x00007ffff541abd7 in __assert_fail_base (fmt=<optimized out>, 
    assertion=assertion@entry=0x7ffff3757b40 "!isAnnotation() && \"Annotation tokens have no length field\"", 
    file=file@entry=0x7ffff3757ad0 "/home/whisperity/large/CodeCompass/CodeCompass-deps/deps/llvm-3.8.0.src/tools/clang/include/clang/Lex/Token.h", 
    line=line@entry=128, 
    function=function@entry=0x7ffff3758660 <clang::Token::getLength() const::__PRETTY_FUNCTION__> "unsigned int clang::Token::getLength() const") at assert.c:92
#3  0x00007ffff541ac82 in __GI___assert_fail (assertion=0x7ffff3757b40 "!isAnnotation() && \"Annotation tokens have no length field\"", 
    file=0x7ffff3757ad0 "/home/whisperity/large/CodeCompass/CodeCompass-deps/deps/llvm-3.8.0.src/tools/clang/include/clang/Lex/Token.h", line=128, 
    function=0x7ffff3758660 <clang::Token::getLength() const::__PRETTY_FUNCTION__> "unsigned int clang::Token::getLength() const") at assert.c:101
#4  0x00007ffff25d3b35 in clang::Token::getLength() const () from /large/whisperity/CodeCompass/CodeCompass-install/bin/../lib/parserplugin/libcppparser.so
#5  0x00007ffff347f008 in clang::Lexer::getSpelling(clang::Token const&, clang::SourceManager const&, clang::LangOptions const&, bool*) ()
   from /large/whisperity/CodeCompass/CodeCompass-install/bin/../lib/parserplugin/libcppparser.so
#6  0x00007ffff246c581 in clang::Preprocessor::getSpelling (this=0x7fff08aeac60, Tok=..., Invalid=0x0)
    at /large/whisperity/CodeCompass/CodeCompass-deps/include/clang/Lex/Preprocessor.h:1346
#7  0x00007ffff246abcb in cc::parser::PPMacroCallback::MacroExpands (this=0x7fff93f5d1c0, macroNameTok_=..., md_=..., range_=...)
    at /home/whisperity/large/CodeCompass/CodeCompass/plugins/cpp/parser/src/ppmacrocallback.cpp:115
#8  0x00007ffff34b8ec2 in clang::PPChainedCallbacks::MacroExpands(clang::Token const&, clang::MacroDefinition const&, clang::SourceRange, clang::MacroArgs const*) ()
   from /large/whisperity/CodeCompass/CodeCompass-install/bin/../lib/parserplugin/libcppparser.so
#9  0x00007ffff34d3b9a in clang::Preprocessor::HandleMacroExpandedIdentifier(clang::Token&, clang::MacroDefinition const&) ()
   from /large/whisperity/CodeCompass/CodeCompass-install/bin/../lib/parserplugin/libcppparser.so
#10 0x00007ffff34fc85f in clang::Preprocessor::HandleIdentifier(clang::Token&) ()
   from /large/whisperity/CodeCompass/CodeCompass-install/bin/../lib/parserplugin/libcppparser.so
#11 0x00007ffff3483341 in clang::Lexer::LexIdentifier(clang::Token&, char const*) ()
   from /large/whisperity/CodeCompass/CodeCompass-install/bin/../lib/parserplugin/libcppparser.so
#12 0x00007ffff3488ae5 in clang::Lexer::LexTokenInternal(clang::Token&, bool) ()
   from /large/whisperity/CodeCompass/CodeCompass-install/bin/../lib/parserplugin/libcppparser.so
#13 0x00007ffff3487c40 in clang::Lexer::Lex(clang::Token&) () from /large/whisperity/CodeCompass/CodeCompass-install/bin/../lib/parserplugin/libcppparser.so
#14 0x00007ffff34fcbaa in clang::Preprocessor::Lex(clang::Token&) () from /large/whisperity/CodeCompass/CodeCompass-install/bin/../lib/parserplugin/libcppparser.so
#15 0x00007ffff2921006 in clang::Parser::ConsumeToken() () from /large/whisperity/CodeCompass/CodeCompass-install/bin/../lib/parserplugin/libcppparser.so
#16 0x00007ffff29264de in clang::Parser::ParseDeclOrFunctionDefInternal(clang::Parser::ParsedAttributesWithRange&, clang::ParsingDeclSpec&, clang::AccessSpecifier)
    () from /large/whisperity/CodeCompass/CodeCompass-install/bin/../lib/parserplugin/libcppparser.so
#17 0x00007ffff2926966 in clang::Parser::ParseDeclarationOrFunctionDefinition(clang::Parser::ParsedAttributesWithRange&, clang::ParsingDeclSpec*, clang::AccessSpecifier) () from /large/whisperity/CodeCompass/CodeCompass-install/bin/../lib/parserplugin/libcppparser.so
#18 0x00007ffff292609f in clang::Parser::ParseExternalDeclaration(clang::Parser::ParsedAttributesWithRange&, clang::ParsingDeclSpec*) ()
   from /large/whisperity/CodeCompass/CodeCompass-install/bin/../lib/parserplugin/libcppparser.so
#19 0x00007ffff2957363 in clang::Parser::ParseInnerNamespace(std::vector<clang::SourceLocation, std::allocator<clang::SourceLocation> >&, std::vector<clang::IdentifierInfo*, std::allocator<clang::IdentifierInfo*> >&, std::vector<clang::SourceLocation, std::allocator<clang::SourceLocation> >&, unsigned int, clang::SourceLocation&, clang::ParsedAttributes&, clang::BalancedDelimiterTracker&) () from /large/whisperity/CodeCompass/CodeCompass-install/bin/../lib/parserplugin/libcppparser.so
#20 0x00007ffff2957198 in clang::Parser::ParseNamespace(unsigned int, clang::SourceLocation&, clang::SourceLocation) ()
   from /large/whisperity/CodeCompass/CodeCompass-install/bin/../lib/parserplugin/libcppparser.so
#21 0x00007ffff2940d86 in clang::Parser::ParseDeclaration(unsigned int, clang::SourceLocation&, clang::Parser::ParsedAttributesWithRange&) ()
   from /large/whisperity/CodeCompass/CodeCompass-install/bin/../lib/parserplugin/libcppparser.so
#22 0x00007ffff2925d2b in clang::Parser::ParseExternalDeclaration(clang::Parser::ParsedAttributesWithRange&, clang::ParsingDeclSpec*) ()
   from /large/whisperity/CodeCompass/CodeCompass-install/bin/../lib/parserplugin/libcppparser.so
#23 0x00007ffff2957363 in clang::Parser::ParseInnerNamespace(std::vector<clang::SourceLocation, std::allocator<clang::SourceLocation> >&, std::vector<clang::IdentifierInfo*, std::allocator<clang::IdentifierInfo*> >&, std::vector<clang::SourceLocation, std::allocator<clang::SourceLocation> >&, unsigned int, clang::SourceLocation&, clang::ParsedAttributes&, clang::BalancedDelimiterTracker&) () from /large/whisperity/CodeCompass/CodeCompass-install/bin/../lib/parserplugin/libcppparser.so
#24 0x00007ffff2957198 in clang::Parser::ParseNamespace(unsigned int, clang::SourceLocation&, clang::SourceLocation) ()
---Type <return> to continue, or q <return> to quit---
```

**Comment by whisperity** [comment]
More explanation about this issue has been presented in #238, and a dummy fix preventing this assertion is in #237.
